iroha-cpp/services/Tx.cpp

```diff
@@ -0,0 +1,383 @@
+{
+    iroha::protocol::CreateAsset pb_create_asset;
+    pb_create_asset.set_asset_name(asset_name);
+    pb_create_asset.set_domain_id(domain_id);
+    pb_create_asset.set_precision(precision);
+
+    auto cmdCreateAsset = iroha::protocol::Command();
+    cmdCreateAsset.set_allocated_create_asset(new iroha::protocol::CreateAsset(pb_create_asset));
+
+    //    auto pl = pbtx.mutable_payload()->mutable_reduced_payload();
+    //    auto cmd = pl->add_commands();
+    //    new (cmd)
+    //        iroha::protocol::Command(cmdCreateAsset);
+    addCommand(cmdCreateAsset);
+    return *this;
+}
+
+Tx& Tx::createAccount(const std::string& account_id, const std::string& domain_id, const std::string& pubkey)
+{
+    iroha::protocol::CreateAccount pb_create_account;
+    pb_create_account.set_account_name(account_id);
+    pb_create_account.set_domain_id(domain_id);
+    pb_create_account.set_public_key(iroha::hexstringToArray<iroha::pubkey_t::size()>(pubkey).value().to_hexstring());
+    // pb_create_account.set_public_key(pubkey); // alternatywa-ok?
+
+    auto cmdCreateAccount = iroha::protocol::Command();
+    cmdCreateAccount.set_allocated_create_account(new iroha::protocol::CreateAccount(pb_create_account));
+
+    addCommand(cmdCreateAccount);
+    return *this;
+}
+
+Tx& Tx::setAccountDetail(const std::string& account_id, const std::string& key, const std::string& value)
+{
+    iroha::protocol::SetAccountDetail cmd;
+    cmd.set_account_id(account_id);
+    cmd.set_key(key);
+    cmd.set_value(value);
+
+    auto cmdSetAccountDetail = iroha::protocol::Command();
+    cmdSetAccountDetail.set_allocated_set_account_detail(new iroha::protocol::SetAccountDetail(cmd));
+
+    addCommand(cmdSetAccountDetail);
+    return *this;
+}
+
+Tx& Tx::compareAndSetAccountDetail(const std::string& account_id, const std::string& key, const std::string& value, const std::optional<std::string>& old_value, bool check_empty)
+{
+    iroha::protocol::CompareAndSetAccountDetail cmd;
+    cmd.set_account_id(account_id);
+    cmd.set_key(key);
+    cmd.set_value(value);
+    if (old_value.has_value())
+        cmd.set_old_value(*std::move(old_value));
+    cmd.set_check_empty(check_empty);
+
+    auto cmdCompareAndSetAccountDetail = iroha::protocol::Command();
+    cmdCompareAndSetAccountDetail.set_allocated_compare_and_set_account_detail(new iroha::protocol::CompareAndSetAccountDetail(cmd));
+
+    addCommand(cmdCompareAndSetAccountDetail);
+    return *this;
+}
+
+Tx& Tx::createDomain(const std::string& domain_id, const std::string& user_default_role)
+{
+    iroha::protocol::CreateDomain pb_create_domain;
+    pb_create_domain.set_domain_id(domain_id);
+    pb_create_domain.set_default_role(user_default_role);
+
+    auto cmdCreateDomain = iroha::protocol::Command();
+    cmdCreateDomain.set_allocated_create_domain(new iroha::protocol::CreateDomain(pb_create_domain));
+
+    //    auto pl = pbtx.mutable_payload()->mutable_reduced_payload();
```

Comment:
the file contains lots of comments to clean up

Reply:
Outdated comment.